pypesto/objective/amici/inner_calculator_collector.py

```diff
@@ -0,0 +1,462 @@
+from __future__ import annotations
```

Comment:
Please add some (short) module-level docstring

Reply:
"""Module for the InnerCalculatorCollector class.

In case of non-quantitative measurements, this class is used to collect
hierarchical inner calculators for each data type and merge their results."""